goldrush_path/goldrush_path.cpp

```diff
@@ -1040,6 +1114,45 @@ main(int argc, char** argv)
     num_and_type_path_log = "the golden path";
   }
 
+  constexpr size_t MEDIAN_SAMPLES_NEEDED = 50000;
+
+  if (opt::phred_min == 0) {
+    std::cerr << "Calculating minimum phred score via median" << std::endl;
+    std::atomic<size_t> num_reads{0};
+    std::vector<uint32_t> phred_scores(MEDIAN_SAMPLES_NEEDED, 0);
+    btllib::SeqReader reader(opt::input, btllib::SeqReader::Flag::LONG_MODE);
+#pragma omp parallel
+  for (const auto record : reader) {
+    if (record.seq.size() < opt::min_length) {
+      continue;
+    }
+    size_t current_index = num_reads.fetch_add(1, std::memory_order_relaxed);
+    if (current_index >= MEDIAN_SAMPLES_NEEDED) {
+      break;
+    }
+
+
+    const auto phred_stat = calc_phred_average(record.qual);
+    phred_scores[current_index] = phred_stat.first;
+
+  }
+  opt::phred_min = calc_median(phred_scores, num_reads);
+  if (opt::debug) {
+    // print out number of reads used to calculate median
+    std::cerr << "Number of reads used to calculate median: " << num_reads << std::endl;
+    // print out median array
+    std::cerr << "Median array: ";
+    for (const auto& phred_score : phred_scores) {
+      std::cerr << phred_score << " ";
+    }

```

Comment:
Probably worth extracting to a separate method to avoid `main()` being too long?

Reply:
done